algo/src/main/java/com/hartwig/serve/sources/ckb/ActionableFunctions.java

```diff
@@ -0,0 +1,102 @@
+import org.apache.logging.log4j.Logger;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+class ActionableFunctions {
+
+    private static final Logger LOGGER = LogManager.getLogger(ActionableFunctions.class);
+
+    @Nullable
+    public static CancerTypeExtraction extractCancerTypeDetails(@NotNull Indication indication) {
+        String[] sourceCancerTypeDetails = extractSourceCancerTypeDetails(indication.termId());
+
+        if (sourceCancerTypeDetails == null) {
+            return null;
+        }
+
+        ImmutableCancerType.Builder applicableCancerTypeBuilder = ImmutableCancerType.builder().name(indication.name());
+        Set<CancerType> blacklistedCancerTypes = Sets.newHashSet();
+
+        assert sourceCancerTypeDetails.length == 2;
+        String source = sourceCancerTypeDetails[0];
+        String id = sourceCancerTypeDetails[1];
+        if (source.equalsIgnoreCase("doid")) {
+            applicableCancerTypeBuilder.doid(id);
+        } else if (source.equalsIgnoreCase("jax")) {
+            switch (id) {
+                case CancerTypeConstants.JAX_ADVANCED_SOLID_TUMOR: {
+                    applicableCancerTypeBuilder.doid(CancerTypeConstants.CANCER_DOID);
+                    blacklistedCancerTypes.add(CancerTypeConstants.LEUKEMIA_TYPE);
+                    blacklistedCancerTypes.add(CancerTypeConstants.REFRACTORY_HEMATOLOGIC_TYPE);
+                    blacklistedCancerTypes.add(CancerTypeConstants.BONE_MARROW_TYPE);
+                    break;
+                }
+                case CancerTypeConstants.JAX_CANCER_OF_UNKNOWN_PRIMARY: {
+                    applicableCancerTypeBuilder.doid(CancerTypeConstants.CANCER_DOID);
+                    break;
+                }
+                case CancerTypeConstants.JAX_CARCINOMA_OF_UNKNOWN_PRIMARY: {
+                    applicableCancerTypeBuilder.doid(CancerTypeConstants.CARCINOMA_OF_UNKNOWN_PRIMARY);
+                    break;
+                }
+                case CancerTypeConstants.JAX_ADENOCARCINOMA_OF_UNKNOWN_PRIMARY: {
+                    applicableCancerTypeBuilder.doid(CancerTypeConstants.ADENOCARCINOMA_OF_UNKNOWN_PRIMARY);
+                    break;
+                }
+                case CancerTypeConstants.JAX_SQUAMOUS_CELL_CARCINOMA_OF_UNKNOWN_PRIMARY: {
+                    applicableCancerTypeBuilder.doid(CancerTypeConstants.SQUAMOUS_CELL_CARCINOMA_OF_UNKNOWN_PRIMARY);
+                    break;
+                }
+                default: {
+                    // CKB uses 10000005 for configuring "Not a cancer". We can ignore these.
+                    if (!id.equals(CancerTypeConstants.JAX_NOT_CANCER)) {
+                        LOGGER.warn("Unexpected DOID string annotated by CKB: '{}'", source + ":" + id);
+                    }
+                    return null;
+                }
+            }
+        } else {
+            LOGGER.warn("Unexpected source '{}'", source);
```

Comment:
SERVE output can be used in case this situation occurs. The WARNs in SERVE are fairly similar to the warns in ACTIN-clinical, ideally they should be captured in a single source and handed differently (not relying on a human reading the command line output). I'd say for now lets leave these WARNs, as long as we have no intention to go full CICD with SERVE